### PR TITLE
 Update `oscar_wagtail` to the last versions of Oscar and Wagtail

### PR DESCRIPTION
**Pull request by samitnuk**
Test with Python 2.7 failed, since Wagtail > 2.0 does not support it.

### CODE REVIEW

**Comment by pjstevns** [comment]
Hi,

Dropping support for python2 seems reasonable. This should be reflected in the tox.ini file.

What I don't understand is the reason behind pinning to to latest versions of wagtail and oscar. Versions in setup.py should be the minimum required versions. Doing otherwise will prevent people from installing this package in say wagtail 2.0 even though the package is perfectly compatible.

Finally, workflow specific exclusions do not belong in .gitignore but in your local .git/info/exclude.

**Comment by samitnuk** [comment]
@pjstevns 

Thanks for your comments! I will modify this PR.

**Comment by samitnuk** [comment]
@pjstevns 

I updated this PR. Please review and comment. Thanks!

File: tests/project/urls.py [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 from oscar.app import application
 
 from oscar_wagtail import urls as oscar_wagtail_urls
 
 admin.autodiscover()
 
 urlpatterns = [
-    url(r'^admin/', include(admin.site.urls)),
-    url(r'', include(application.urls)),
+    url(r'^admin/', admin.site.urls),
```
Comment (pjstevns) [inline tests/project/urls.py:12]: why are you removing the include wrappers here?

File: tests/project/urls.py [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
+    url(r'', application.urls),
     url(r'^i18n/', include('django.conf.urls.i18n')),
 
     url(r'^dashboard/cms/', include(wagtailadmin_urls.urlpatterns)),
     url(r'^dashboard/cms/',
-        include(oscar_wagtail_urls.urlpatterns, namespace='oscar_wagtail'))
+        include((oscar_wagtail_urls.urlpatterns, 'oscar_wagtail'),
```
Comment (pjstevns) [inline tests/project/urls.py]: what is the reason for this change. It doesnt look right.

File: tests/project/urls.py (hunk 1 above)
Comment (samitnuk) [reply tests/project/urls.py:12]: This is required for Django >=1.9. Explained [here].

[here]: https://docs.djangoproject.com/en/dev/releases/1.9/#passing-a-3-tuple-or-an-app-name-to-include.

File: tests/project/urls.py (hunk 2 above)
Comment (samitnuk) [reply tests/project/urls.py]: This is also explained [here]. But I revised this part one more time.

[here]: https://docs.djangoproject.com/en/dev/releases/1.9/#passing-a-3-tuple-or-an-app-name-to-include.

File: tests/project/urls.py (hunk 1 above)
Comment (pjstevns) [reply tests/project/urls.py:12]: I think you are misreading the docs there. The include wrapper is required. Just keep the code as it was.

Comment (samitnuk) [reply tests/project/urls.py:12]: Sorry, but this part does not work with the `include` wrapper.

And this what we have in the [docs]:
img 2018-12-10 18 56 55

If you run tests with `url(r'^admin/', include(admin.site.urls)),` you will get the next [traceback]. This is the last part of this traceback (with better highlighting):
img 2018-12-10 19 09 46

This part must be changed (we must remove `include` here).

[docs]: https://docs.djangoproject.com/en/dev/releases/1.9/#passing-a-3-tuple-or-an-app-name-to-include
[traceback]: https://gist.github.com/samitnuk/7c48927f2ab67f8b94cb191939d5210c

**Comment by thqvlg** [comment]
There was a error, when access the url "http://127.0.0.1:8000/dashboard/catalogue/categories/"
OperationalError at /dashboard/catalogue/categories/
no such column: catalogue_category.page_ptr_id

python3.6.7
Django2.1.2
wagtail2.4

How to resolve it? thanks

**Comment by thqvlg** [comment]
I got a error when create a page.

OperationalError at /cms/pages/add/catalogue/category/2/
foreign key mismatch - "offer_range_included_categories" referencing "catalogue_category"
Request Method:	POST
Request URL:	http://127.0.0.1:8000/cms/pages/add/catalogue/category/2/
Django Version:	2.1.2
Exception Type:	OperationalError
Exception Value:	foreign key mismatch - "offer_range_included_categories" referencing "catalogue_category"
Exception Location:	Proj\.env\lib\site-packages\django\db\backends\sqlite3\base.py in execute, line 296
Python Executable:	Proj\.env\Scripts\python.exe
Python Version:	3.6.7

wagtail is 2.4


when create category from dashboard, i got another error:

ValidationError at /dashboard/catalogue/categories/create/
{'title': ['此字段不能为空。'], 'slug': ['此字段不能为空。'], 'path': ['具有 Path 的 Page 已存在。']}
Request Method:	POST
Request URL:	http://127.0.0.1:8000/dashboard/catalogue/categories/create/
Django Version:	2.1.2
Exception Type:	ValidationError
Exception Value:	
{'title': ['此字段不能为空。'], 'slug': ['此字段不能为空。'], 'path': ['具有 Path 的 Page 已存在。']}
Exception Location:	Proj\.env\lib\site-packages\django\db\models\base.py in full_clean, line 1152
Python Executable:	Proj\.env\Scripts\python.exe
Python Version:	3.6.7
